src/typing/flow_js.ml

```diff
@@ -5352,6 +5353,17 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
       rec_flow cx trace (key, ElemT (unknown_use, reason_lookup, l, action))
 
 
+    | UnionT (_, rep),
+      ElemT (use_op, reason, tin, ReadElem tout) ->
+      let t0, (t1, ts) = UnionRep.members_nel rep in
+      let f t =
+        AnnotT (reason, Tvar.mk_where cx reason (fun tvar ->
```

Comment:
@samwgoldman: Removing this AnnotT still passes the tests, [including the one that surfaced internally](https://flow.org/try/#0C4TwDgpgBAYglhANgEwM5QLxQN4B8BQUUAhgFxSrABOcAdgOYA0hUARubQK4C2rEVzXAF8A3PnwBjAPa1KUZMWBlYCFOizYWygOQAzKVO3Mi7KABYATM1HjdnWhOBwZUehGDwkyADwBpcgAkvhAgqN6eagB8kQAUANYh5L4AlIEAoogQ3BC0wAAq4BDhqmiMUL6ROCxU7pxUtPKKxADaCSAAumJC4tKywCQAasSInNBYbh4lMdrE2slivXKsQyNjru4RyNOscyJAA)

Going back to the conversation we had two days ago, I think that the UnionT's tvars should "in theory" all be 0->1 anyway, so distributing ElemT through the union should preserve the 0->1-ness. On the other hand, we know that doesn't hold in practice, so maybe goodmind should add a field like DestructAnnot vs. DestructInfer to ElemT and only use this special case on `Annot`.

Reply:
Should I wait for @samwgoldman answer or add this field now?

Reply:
Wait a few days first, but let's say if Sam doesn't answer by Monday then yes! I just want to give him a fair amount of time to take a look here first.